infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/DefaultClusterManager.java

```diff
@@ -137,97 +124,67 @@ public void shutdown() {
         log.info("Shutdown Cluster Manager completed.");
     }
 
-    public static TopologyDescriptor readConfig() {
-        ClusterDescriptor activeCluster;
-        List<String> activeNodeNames = new ArrayList<>();
-        List<String> standbyNodeNames = new ArrayList<>();
-        List<String> activeNodeHosts = new ArrayList<>();
-        List<String> standbyNodeHosts = new ArrayList<>();
-        List<String> activeNodeIds = new ArrayList<>();
-        List<String> standbyNodeIds = new ArrayList<>();
-        String activeClusterId;
-        String activeCorfuPort;
-        String activeLogReplicationPort;
-
-        String standbySiteName;
-        String standbyCorfuPort;
-        String standbyLogReplicationPort;
-
-        File configFile = new File(CONFIG_FILE_PATH);
-        try (FileReader reader = new FileReader(configFile)) {
-            Properties props = new Properties();
-            props.load(reader);
-
-            Set<String> names = props.stringPropertyNames();
-
-            activeClusterId = props.getProperty(ACTIVE_CLUSTER_NAME, DEFAULT_ACTIVE_CLUSTER_NAME);
-            activeCorfuPort = props.getProperty(ACTIVE_CLUSTER_CORFU_PORT);
-            activeLogReplicationPort = props.getProperty(LOG_REPLICATION_SERVICE_ACTIVE_PORT_NUM);
-            for (int i = 0; i < NUM_NODES_PER_CLUSTER; i++) {
-                String nodeName = ACTIVE_CLUSTER_NODE + i;
-                if (!names.contains(nodeName)) {
-                    continue;
-                }
-                activeNodeNames.add(nodeName);
-                activeNodeHosts.add(props.getProperty(nodeName));
+    public TopologyDescriptor readConfig() {
+        List<ClusterDescriptor> activeClusters = new ArrayList<>();
+        List<ClusterDescriptor> standbyClusters = new ArrayList<>();
+
+        List<String> activeClusterIds = topology.getActiveClusterIds();
+        List<String> activeCorfuPorts = topology.getActiveCorfuPorts();
+        List<String> activeLogReplicationPorts =
+            topology.getActiveLogReplicationPorts();
+        List<String> activeNodeNames = topology.getActiveNodeNames();
+        List<String> activeNodeHosts = topology.getActiveIpAddresses();
+        List<String> activeNodeIds = topology.getActiveNodeUuids();
+
+        List<String> standbyClusterIds = topology.getStandbyClusterIds();
+        List<String> standbyCorfuPorts = topology.getStandbyCorfuPorts();
+        List<String> standbyLogReplicationPorts =
+            topology.getStandbyLogReplicationPorts();
+        List<String> standbyNodeNames = topology.getActiveNodeNames();
+        List<String> standbyNodeHosts = topology.getStandbyIpAddresses();
+        List<String> standbyNodeIds = topology.getStandbyNodeUuids();
+
```

Comment:
Actually, there will no change for the existing tests because we only create a single source/sink site.  The received topology will have multiple source/sink clusters but only 1 will be functional.